website/docs/plugin/log/filtering.mdx

```diff
@@ -0,0 +1,187 @@
+---
+page_title: 'Home - Plugin Development: Filtering Log Output'
+description: >-
+  Omit or mask specific log information from your provider to hide sensitive data.
+---
+
+# Filtering Log Output
+
+When providers [write logs](/plugin/log/writing), there may be sensitive data which should not be present in log messages or structured log fields. The [`tflog` package](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-log/tflog) supports masking data or omitting log entries entirely before they are output via the provider root logger or provider-defined subsystem loggers.
+
+~> **NOTE**: While log filtering can help hide sensitive data, it is important to ensure the provider implementation works as expected before creating production provider releases.
+
+## Masking Log Output
+
+Masking data in log messages or structured log fields is the process of replacing a sensitive piece of data with a placeholder piece of data. The `tflog` package uses `***` as the replacement.
```

Comment:
Btw, I know is of little to no importance, but that `***` choice was entirely arbitrary and, if you think a better one can be used, we should definitely change it.

Initially I wanted to replace each character in the string with `*`, but it felt like sharing the length of a secret was also a bad idea.

We could have gone with `<REDACTED>`? 🤷‍♀️ 

Happy to leave it as is, and also to change it too.

Reply:
I personally think `***` is an excellent choice for masking, similar to how other products like GitHub Actions log masking works. 👍 Agreed about making it a static length. Using a word like "redacted"/"omitted"/"elided" etc implies English-based logging, which may not always be the case. So yeah, I think its all good!